client/docs/plugin_model.md

```diff
@@ -0,0 +1,169 @@
+<!---
+Licensed under Creative Commons Attribution 4.0 International License
+https://creativecommons.org/licenses/by/4.0/
+--->
+
+# Building Contract Plugins #
+
+The plugins module provides a means to build and use operations that
+is independent (as much as possible) of the actual invocation of the
+operation. For example, the `mock-contract` plugin wraps the
```

Comment:
Does this PR have a plugin for mock-contract? I am afraid I only see attestion-test and kv-test plugins?

Reply:
it doesn't. we could write one but the other test contracts are so simple that it doesnt make much sense.